pkg/controller/migmigration/backup.go

```diff
@@ -456,10 +456,15 @@ func (t *Task) buildBackup(client k8sclient.Client) (*velero.Backup, error) {
 	if err != nil {
 		return nil, liberr.Wrap(err)
 	}
+
+	// Construct a bakcup name like "$migrationname-initial-" or "$migrationname-stage-".
+	// Truncate the name to 57 chars to leave room for 5 char generateName suffix. Limit is 63 chars.
+	truncatedGenerateName := fmt.Sprintf("%.57s-", fmt.Sprintf("%s-%s", t.Owner.GetName(), backupTypePrefix))
```

Comment:
It looks like we are truncating the end product of string concatenation. Can we truncate the `t.Owner.GetName()` alone so we never truncate the suffix `stage` or `initial` since we know the length of these strings beforehand, we can truncate the migration name to 50 chars `57 - max('stage', 'initial')`

Reply:
Good idea, will change.

Reply:
@pranavgaikwad changed. This generates a 57 char string now. 

If you'd like to try it out with some examples. https://play.golang.org/p/coW_c-fUnJ0

Also note that generateName gracefully handles names > 63 chars, it just truncates on its own. I verified against my  cluster.